internal/pkg/cli/deploy/env.go

```diff
@@ -140,6 +142,10 @@ func NewEnvDeployer(in *NewEnvDeployerInput) (*envDeployer, error) {
 	if err != nil {
 		return nil, fmt.Errorf("initialize env describer: %w", err)
 	}
+	overrider := in.Overrider
+	if overrider == nil {
+		overrider = new(override.Noop)
+	}
```

Comment:
sorry late to the party! Is there a reason why this is treated differently than workload deployer? https://github.com/aws/copilot-cli/blob/feb5084f180e8a0fe3dc173aa37be39944ca9234/internal/pkg/cli/deploy/workload.go#L240

Reply:
Oh! I was supposed to remove the if-statement in the follow-up https://github.com/aws/copilot-cli/pull/4428.

I had it here so that the code didn't panic until I added the code that initialized the overrider, does that make sense?
I'll remove it in my next PR!